docs/app-check/debug-provider.md

```diff
@@ -16,9 +18,135 @@ Warning: The debug provider allows access to your Firebase resources from
+    set the App Check debug provider factory in Debug builds:
+
+    * {Swift}
+
+      ```swift
+      import UIKit
+      import Flutter
+      import Firebase  // Add the Firebase import.
+
+      @UIApplicationMain
+      @objc class AppDelegate: FlutterAppDelegate {
+        override func application(
+          _ application: UIApplication,
+          didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
+        ) -> Bool {
+          // Use the debug provider in Debug builds:
+      #if DEBUG
+          let providerFactory = AppCheckDebugProviderFactory()
+          AppCheck.setAppCheckProviderFactory(providerFactory)
+      #endif
+
+          GeneratedPluginRegistrant.register(with: self)
+          return super.application(application, didFinishLaunchingWithOptions: launchOptions)
+        }
+      }
+      ```
+
+      Tip: Confirm in your Xcode project's Build Settings that it passes the
+      `DEBUG` flag to the Swift compiler.
+
+    * {Objective-C}
+
+      ```objc
+      #import "AppDelegate.h"
+      #import "GeneratedPluginRegistrant.h"
+      @import Firebase;  // Add the Firebase import.
+
+      @implementation AppDelegate
+
+      - (BOOL)application:(UIApplication *)application
+          didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+        // Use the debug provider in Debug builds:
+      #if DEBUG
+        FIRAppCheckDebugProviderFactory *providerFactory = [[FIRAppCheckDebugProviderFactory alloc] init];
+        [FIRAppCheck setAppCheckProviderFactory:providerFactory];
+      #endif
+        [GeneratedPluginRegistrant registerWithRegistry:self];
+        // Override point for customization after application launch.
+        return [super application:application didFinishLaunchingWithOptions:launchOptions];
+      }
+
+      @end
+      ```
+
+1.  Enable debug logging in your Xcode project (v11.0 or newer):
+
+    1.  Open **Product > Scheme > Edit scheme**.
+    1.  Select **Run** from the left menu, then select the **Arguments** tab.
+    1.  In the **Arguments Passed on Launch** section, add `-FIRDebugEnabled`.
+
+1.  Open `ios/Runner.xcworkspace` with Xcode and run your app in the Simulator.
+    A local debug token will be logged to the debug output when Firebase tries
+    to send a request to the backend. For example:
+
+    <pre>[Firebase/AppCheck][I-FAA001001] Firebase App Check Debug Token:
+    123a4567-b89c-12d3-e456-789012345678</pre>
+
+{# Google-internal common file: #}
+<<../_includes/manage-debug-tokens.md>>
+
+After you register the token, Firebase backend services will accept it as valid.
+
+Because this token allows access to your Firebase resources without a
+valid device, it is crucial that you keep it private. Don't commit it to a
+public repository, and if a registered token is ever compromised, revoke it
+immediately in the Firebase console.
+
+## Android
+
+The debug provider is not currently supported in Flutter apps on Android. See
```

Comment:
I think we will be updating the [android implementation](https://github.com/firebase/flutterfire/pull/9412#discussion_r963419690) so the user can use the debug provider on android. So we'll get that updated first, and then update this part of the docs. Other than that, it LGTM 👍